### PR TITLE
Add location to listings

### PR DESCRIPTION
**Pull request by mariocsee**
## What type of PR is this? (check all applicable)

- [x] Refactor

## Description
Add location to listings. This should be helpful for events or job listings in certain locations. (Later, add searching and filtering based on location.)

## Related Tickets & Documents
#3425 

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
To be added.
Screen Shot 2019-07-02 at 17 14 52
Screen Shot 2019-07-02 at 17 14 35
Screen Shot 2019-07-02 at 17 14 28

## Added to documentation?

- [x] no documentation needed

### CODE REVIEW

File: app/views/classified_listings/_form.html.erb [hunk 1]
```diff
@@ -55,6 +55,10 @@
       </div>
     </div>
       <%= javascript_pack_tag "listingForm", defer: true %>
+    <div class="field">
+      <%= form.label "location", "Location" %>
+      <%= form.text_field "location", placeholder: "64 characters max, plain text" %>
```
Comment (rhymes) [inline app/views/classified_listings/_form.html.erb]: this is not validated in the model, so in theory I could add as many chars as I want to, would it break the UI?

Comment (mariocsee) [reply app/views/classified_listings/_form.html.erb]: Good catch! I should probably limit it in the model as well to keep things short. If the location is long it pushes to the edit/report abuse buttons to a new line, so I could look into dedicating a line just for location (if provided of course) as to not mess with the placement of action buttons.

**Comment by mariocsee** [comment]
@rhymes Thanks for the review! Yeah the implementation right now is bare minimum with just a string to describe location. 

I think improvements to grouping multiple ways to refer to the same place together will come a bit later as we flesh out more details in listings world. It would definitely be easier to search "NYC" and have all listings with "New York, NY", "Manhattan", etc. show up.

**Comment by mariocsee** [comment]
@benhalpern I just added the length validation in the listings model, and I definitely agree with steps moving forward in adding more sophistication with location details and maximizing Algolia's services. 

It'll take me a while to get these implemented with other PRs I'm working on, so I think merging this for now is fine. We can open a new issue with what you just outlined as well to keep tabs on additions to location.

**Pull request merged**